framework/src/functions/LinearCombinationFunction.C

```diff
@@ -61,3 +61,21 @@ LinearCombinationFunction::value(Real t, const Point & p)
     val += _w[i] * _f[i]->value(t, p);
   return val;
 }
+
+RealGradient
+LinearCombinationFunction::gradient(Real t, const Point & p)
+{
+  RealGradient g;
+  for (unsigned i = 0; i < _f.size(); ++i)
+    g += _w[i] * _f[i]->gradient(t, p);
+  return g;
+}
+
+RealVectorValue
+LinearCombinationFunction::vectorValue(Real t, const Point & p)
```

Comment:
So having both isn't a problem except it's code duplication 😛 . If you really want to fix that, you'd call a templated helper function:

```
template <typename T>
T vectorHelper(Real t, const Point & p)
{ 
  // One copy of logic here ...
}

RealVectorValue
LinearCombinationFunction::vectorValue(Real t, const Point & p)
{
    return vectorHelper<RealVectorValue>(t, P);
}

Gradient
LinearCombinationFunction::gradient(Real t, const Point & p)
{
    return vectorHelper<Gradient>(t, P);
}
```